### PR TITLE
CI: Setup Cilium with Hubble Enabled nightly run

### PR DESCRIPTION
**Pull request by MikeZappa87**
**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by matmerr — APPROVED** [review]

**Comment by MikeZappa87** [comment]
@matmerr take a look when you get a chance. Make sure I covered everything.

File: test/integration/manifests/cilium/v1.14.4/cilium-config/cilium-config-hubble.yaml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  enable-metrics: "true"
+  enable-policy: default
+  enable-remote-node-identity: "true"
+  enable-session-affinity: "true"
+  enable-svc-source-range-check: "true"
+  enable-vtep: "false"
+  enable-well-known-identities: "false"
+  enable-xt-socket-fallback: "true"
+  hubble-listen-address: ""
+  hubble-metrics: flow:sourceContext=pod;destinationContext=pod tcp:sourceContext=pod;destinationContext=pod
```
Comment (MikeZappa87) [inline test/integration/manifests/cilium/v1.14.4/cilium-config/cilium-config-hubble.yaml:46]: @matmerr I think you mentioned we needed additional labels for some of the metrics.

File: test/integration/networkobservability/hubble_test.go [hunk 2]
```diff
@@ -25,13 +27,190 @@ const (
...
+	flow := []string{"source", "destination"}
+	drop := []string{"source", "source"}
+
+	hubbleKeys["hubble_flows_processed_total"] = flow
+	hubbleKeys["hubble_tcp_flags_total"] = flow
+	hubbleKeys["hubble_dns_responses_total"] = dns
+	hubbleKeys["hubble_dns_response_types_total"] = dns
+	hubbleKeys["hubble_dns_queries_total"] = dns
+	hubbleKeys["hubble_drop_total"] = drop
+}
```
Comment (matmerr) [inline test/integration/networkobservability/hubble_test.go:45]: given this init is var instantiation, we could move this to the var() block where it's declared and avoid the additional init

```suggestion

var (
    flow       = []string{"source", "destination"}
    hubbleKeys = map[string][]string{
		    "hubble_flows_processed_total": flow,
             ...
    }
)
```

**Review by matmerr — CHANGES_REQUESTED** [review]
minor things, other than that looks good 🙌

**Pull request closed without merging**